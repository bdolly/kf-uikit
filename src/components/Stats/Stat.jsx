/* eslint-disable */
import React, { Fragment } from 'react';
import propTypes from 'prop-types';
import classes from 'classnames';
import Icon from '../Icon/Icon';

const Stat = ({ icon, label, metric, metricFormatter }) => {
  return (
    <div className="Stat Stat--container">
      <Icon className="Stat--icon" {...label} kind={icon} />
      <div className="Stat--metric">{metricFormatter ? metricFormatter(metric) : metric}</div>
      {label &&
        <div className="Stat--label">{label}</div>
      }
    </div>
  );
};

Stat.propTypes = {
  /** Name of Entity  */
  label: propTypes.string,
  /** 'kind' prop for Icon compoennt */
  icon: propTypes.string.isRequired,
  /** number to display */
  metric: propTypes.number.isRequired,
  /** function that receives and returns the stat number */
  metricFormatter: propTypes.function,
};

Stat.defaultProps = {
  label: null,
  metricFormatter: null,
};

export default Stat;
