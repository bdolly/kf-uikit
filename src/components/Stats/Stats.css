/* Stats Bar */
.Stats--container {
  @apply flex
    items-center
    bg-white
    py-20
    border
    border-solid 
    border-grey-lightest;
}

.Stats--container.transparent {
  @apply bg-transparent
    border-0;
}

.Stats--container-divider {
  @apply border-l
    border-grey-lightest
    h-8;
}

.Stat.Stat--container {
  @apply flex
    flex-1
    justify-center
    items-center;
}

.Stat--icon {
  @apply mr-4;
}
.Stat--metric {
  @apply mr-4
    font-bold 
    text-xl
    text-grey-dark
    tracking-wide;
}
/*TODO: make text trucate when it gets small*/
.Stat--label {
  @apply pt-1
    font-light
    text-sm
    text-grey-light;
}
.Stats--container.small .Stat--container .Stat--metric {
  @apply text-sm;
}
.Stats--container.small .Stat--container .Stat--label {
  @apply hidden;
}
