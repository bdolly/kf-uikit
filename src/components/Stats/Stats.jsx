/* eslint-disable */
import React, { Fragment } from 'react';
import propTypes from 'prop-types';
import classes from 'classnames';
import Stat from './Stat';

const Stats = ({ stats, className, small, transparent }) => {
  const StatsClasses = classes(
    ['Stats--container', 'flex', 'items-center'],
    { small: small, transparent: transparent },
    className,
  );
  return (
    <div className={StatsClasses}>
      {stats.map((stat, i) => (
        <Fragment key={stat.label}>
          {i > 0 && <div key={i} className="Stats--container-divider" />}
          <Stat
            icon={stat.icon}
            label={stat.label}
            metric={stat.metric}
          />
        </Fragment>
      ))}
    </div>
  );
};

Stats.propTypes = {
  /** Array of Stat/Metrics objects to display   */
  stats: propTypes.arrayOf(propTypes.shape({
    /** The icon used for the stat */
    icon: propTypes.string.isRequired,
    /** The label to describe the stat */
    label: propTypes.string,
    /** The actual stat metric */
    metric: propTypes.number.isRequired,
  })),
  /** if set, displays only the icon and metric */
  small: propTypes.bool,
  /** if set, hides the border and background of the Stats-container */
  transparent: propTypes.bool,
};

Stats.defaultProps = {
  stats: [],
  small: false,
  transparent: false,
}
export default Stats;
