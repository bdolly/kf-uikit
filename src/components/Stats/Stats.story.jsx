/* eslint-disable */
import React from 'react';
import { storiesOf } from '@storybook/react';
import { object, boolean, text, number, select } from '@storybook/addon-knobs';
import Stats from './Stats';
import Stat from './Stat';
import iconNames from '../Icon/Icons';

const names = Object.keys(iconNames).reduce((cur, acc) => ({ [cur]: cur, ...acc }), {});
console.log(names);
const stories = storiesOf('Stats', module);

stories.add(
  'Stats Bar',
  () => (
    <Stats
      small={boolean('small', false)}
      transparent={boolean('transparent', false)}
      stats={object(
        'stats',
        [
          { icon: 'file', label: 'Files', metric: 1000 },
          { icon: 'participant', label: 'Participants', metric: 1000 },
          { icon: 'family', label: 'Families', metric: 1000 },
          { icon: 'file-size', label: 'Size', metric: 1000 },
        ],
      )}
    />
  ),
  {
    info: {
      text: `Entity Statistic Bar display a metric (count) for each passed in entity`,
    },
  },
);

stories.add(
  'Stat',
  () => (
    <Stat
      {...{
        icon: select('icon', Object.keys(iconNames)),
        label: text('lable', 'Participants'),
        metric: number('metic', 1000),
      }}
    />
  ),
  {
    info: {
      text: `Single Entity Statistic Display`,
    },
  },
);
