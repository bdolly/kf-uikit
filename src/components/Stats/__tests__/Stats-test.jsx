import React from 'react';
import renderer from 'react-test-renderer';
import Stats from '../Stats';

[true, false].map((small) => {
  [true, false].map((transparent) => {
    [0,1,2,3].map((n) => {
      it(`${small ? 'small' : ''} ${transparent ? 'transparent' : ''} stats bar with ${n} metrics renders correctly`, () => {
          const tree = renderer.create(
                  <Stats
                    small={small}
                    transparent={transparent}
                    stats={Array(n).map((n, i) => ({
                      icon: 'Participant',
                      label: 'Participants',
                      metric: i,
                    }))}
                  />
                  ).toJSON();
            expect(tree).toMatchSnapshot();
      });
    });
  });
});
