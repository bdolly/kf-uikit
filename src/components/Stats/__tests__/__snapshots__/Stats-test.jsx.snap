// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`  stats bar with 0 metrics renders correctly 1`] = `
<div
  className="Stats--container flex items-center"
/>
`;

exports[`  stats bar with 1 metrics renders correctly 1`] = `
<div
  className="Stats--container flex items-center"
/>
`;

exports[`  stats bar with 2 metrics renders correctly 1`] = `
<div
  className="Stats--container flex items-center"
/>
`;

exports[`  stats bar with 3 metrics renders correctly 1`] = `
<div
  className="Stats--container flex items-center"
/>
`;

exports[` transparent stats bar with 0 metrics renders correctly 1`] = `
<div
  className="Stats--container flex items-center transparent"
/>
`;

exports[` transparent stats bar with 1 metrics renders correctly 1`] = `
<div
  className="Stats--container flex items-center transparent"
/>
`;

exports[` transparent stats bar with 2 metrics renders correctly 1`] = `
<div
  className="Stats--container flex items-center transparent"
/>
`;

exports[` transparent stats bar with 3 metrics renders correctly 1`] = `
<div
  className="Stats--container flex items-center transparent"
/>
`;

exports[`small  stats bar with 0 metrics renders correctly 1`] = `
<div
  className="Stats--container flex items-center small"
/>
`;

exports[`small  stats bar with 1 metrics renders correctly 1`] = `
<div
  className="Stats--container flex items-center small"
/>
`;

exports[`small  stats bar with 2 metrics renders correctly 1`] = `
<div
  className="Stats--container flex items-center small"
/>
`;

exports[`small  stats bar with 3 metrics renders correctly 1`] = `
<div
  className="Stats--container flex items-center small"
/>
`;

exports[`small transparent stats bar with 0 metrics renders correctly 1`] = `
<div
  className="Stats--container flex items-center small transparent"
/>
`;

exports[`small transparent stats bar with 1 metrics renders correctly 1`] = `
<div
  className="Stats--container flex items-center small transparent"
/>
`;

exports[`small transparent stats bar with 2 metrics renders correctly 1`] = `
<div
  className="Stats--container flex items-center small transparent"
/>
`;

exports[`small transparent stats bar with 3 metrics renders correctly 1`] = `
<div
  className="Stats--container flex items-center small transparent"
/>
`;
